Source/ExecutionEngine/Caching/VerificationResultCache.cs

```diff
@@ -0,0 +1,86 @@
+﻿#nullable enable
+using System;
+using System.Diagnostics.Contracts;
+using System.Runtime.Caching;
+using System.Text.RegularExpressions;
+using VC;
+
+namespace Microsoft.Boogie
+{
+  public sealed class VerificationResultCache : IVerificationResultCache
+  {
+    private readonly MemoryCache programCache = new("ProgramCache");
+    private readonly MemoryCache Cache = new("VerificationResultCache");
+
+    public Program? CachedProgram(string programId) {
+      return programCache.Get(programId) as Program;
+    }
+
+    private readonly CacheItemPolicy Policy = new CacheItemPolicy
+      {SlidingExpiration = new TimeSpan(0, 10, 0), Priority = CacheItemPriority.Default};
+
+    public void Insert(Implementation impl, ImplementationRunResult result)
+    {
+      Contract.Requires(result != null);
```

Comment:
```suggestion
      Contract.Requires(impl != null);
      Contract.Requires(result != null);
```
This contract was left out during refactoring. Or did you intend to remove both contracts?
The function Lookup, RemoveMatchingKeys and VerificationPriority below also had a contract removed.

Reply:
The presence of the contract creates warning, because it implies `impl` can be `null` and this file uses `#nullable enable`, so I had to remove the check, otherwise the build fails.